pkg/util/labels_test.go

```diff
@@ -0,0 +1,35 @@
+package util
+
+import (
+	"testing"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLabelMatchersToString(t *testing.T) {
+	tests := []struct {
+		input    []*labels.Matcher
+		expected string
+	}{
+		{
+			input:    nil,
+			expected: "{}",
+		}, {
+			input: []*labels.Matcher{
+				labels.MustNewMatcher(labels.MatchEqual, "foo", "bar"),
+			},
+			expected: `{foo="bar"}`,
+		}, {
+			input: []*labels.Matcher{
+				labels.MustNewMatcher(labels.MatchEqual, "foo", "bar"),
+				labels.MustNewMatcher(labels.MatchNotEqual, "who", "boh"),
+			},
+			expected: `{foo="bar",who!="boh"}`,
```

Comment:
Test with `__name__` label name would show that no special handling of that name is done.